File: iFalcon.cpp
class Assignment {
public:
    bool isValid(string s)
    {
        stack<char>Stack;
        for (auto &ch : s)
        {
            if (ch == '(' or ch == '{' or ch == '[')
            {
                Stack.push(ch);
            }

            else
            {
                if (Stack.empty())return false;
                if (  (Stack.top() == '(' and ch == ')')   or  (Stack.top() == '{' and ch == '}') or (Stack.top() == '[' and ch == ']'))
                {
                    Stack.pop();
                }
                else return false;
            }
        }

        return Stack.empty();
    }
};


// The code with main function

#include <bits/stdc++.h>
using namespace std;

bool isValid(string s)
{
    stack<char> Stack;
    for (auto &ch : s)
    {
        if (ch == '(' or ch == '{' or ch == '[')
        {
            Stack.push(ch);
        }

        else
        {
            if (Stack.empty())
                return false;
            if ((Stack.top() == '(' and ch == ')') or (Stack.top() == '{' and ch == '}') or (Stack.top() == '[' and ch == ']'))
            {
                Stack.pop();
            }
            else
                return false;
        }
    }

    return Stack.empty();
}

int main()
{
    int numberOfTestcases;
    cout<<"Enter the number of testcases you want to test: ";
    cin >> numberOfTestcases;

    while (numberOfTestcases--)
    {
        string s;
        cout << "Enter the input string: ";
        cin >> s;

        if (isValid(s))
        {
            cout << "The input string is valid" << endl;
        }
        else
        {
            cout << "The input string is invalid" << endl;
        }
    }
    return 0;
}

File: iFalcon.java
// Code with the main function 

// * Make a file: ParenthesesValidation.java
// * Paste the following code in it.
// * To test the code kindly read the README.md file.

import java.util.Scanner;
import java.util.Stack;

public class ParenthesesValidation {
    public static boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        for (char ch : s.toCharArray()) {
            if (ch == '(' || ch == '{' || ch == '[') {
                stack.push(ch);
            } else {
                if (stack.empty())
                    return false;
                if ((stack.peek() == '(' && ch == ')') || (stack.peek() == '{' && ch == '}') || (stack.peek() == '[' && ch == ']')) {
                    stack.pop();
                } else
                    return false;
            }
        }
        return stack.empty();
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the number of test cases you want to test: ");
        int numberOfTestcases = scanner.nextInt();

        while (numberOfTestcases-- > 0) {
            System.out.print("Enter the input string: ");
            String inputString = scanner.next();

            if (isValid(inputString)) {
                System.out.println("The input string is valid");
            } else {
                System.out.println("The input string is invalid");
            }
        }

        scanner.close();
    }
}


